src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -127,7 +130,17 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             try
             {
                 this.TelemetryClient.Flush();
-                asyncContinuation(null);
+                if (DateTime.UtcNow.AddSeconds(-30) > this.lastLogEventTime)
+                {
+                    // Nothing has been written, so nothing to wait for
+                    asyncContinuation(null);
+                }
+                else
+                {
+                    // Documentation says it is important to wait after flush, else nothing will happen
+                    // https://docs.microsoft.com/en-us/azure/application-insights/app-insights-api-custom-events-metrics#flushing-data
+                    System.Threading.Tasks.Task.Delay(500).ContinueWith((task) => asyncContinuation(null));
```

Comment:
Please use `TimeSpan.FromMilliseconds(500)` to make code a little bit more self-explanatory

Reply:
Fixed